File: Taller/Consumidores/Python/consumer.py
import os
import pika
import sys
rabbitmq_host = os.getenv("RABBITMQ_HOST")
QUEUE_NAME = "hello"

def callback(ch, method, properties, body):
    print(f"[x] Recibido: {body.decode()}", flush=True)  # Forzar salida inmediata

def main():
    try:
        # Configurar conexión a RabbitMQ
        connection = pika.BlockingConnection(pika.ConnectionParameters(host=rabbitmq_host))
        channel = connection.channel()

        # Declarar la cola
        channel.queue_declare(queue=QUEUE_NAME, durable=False)

        print("[*] Esperando mensajes. Para salir, presiona CTRL+C", flush=True)

        # Consumir mensajes
        channel.basic_consume(queue=QUEUE_NAME, on_message_callback=callback, auto_ack=True)

        # Evitar buffering de salida
        sys.stdout.flush()
        channel.start_consuming()

    except Exception as e:
        print(f"Error al conectar con RabbitMQ: {e}", flush=True)

if __name__ == "__main__":
    main()

File: Taller/Consumidores/JS/consumer.js
const amqp = require('amqplib');
const rabbitmqHost = process.env.RABBITMQ_HOST;
async function consume() {
    const connection = await amqp.connect(`amqp://${rabbitmqHost}`);
    const channel = await connection.createChannel();

    await channel.assertQueue("hello", { durable: false });

    console.log(" [*] Esperando mensajes en la cola 'hello'. Ctrl+C para salir.");

    channel.consume("hello", (msg) => {
        console.log(` [x] Recibido: ${msg.content.toString()}`);
    }, { noAck: true });
}

consume().catch(console.error);
